packages/blitz/src/cli/utils/codegen-tasks.ts

```diff
@@ -39,33 +39,23 @@ export const codegenTasks = async () => {
     )
 
     if (hasPrisma) {
-        const tempDir = os.tmpdir()
-        const creationDate = await fs.stat(process.cwd())
-        if (!fs.existsSync(join(tempDir, `blitz-${creationDate.birthtimeMs}-prisma-client.json`))) {
-            const lastModified = await fs.stat(join(process.cwd(), "db/schema.prisma"))
-            await fs.writeFile(
-                join(tempDir, `blitz-${creationDate.birthtimeMs}-prisma-client.json`),
-                JSON.stringify({ lastModified: lastModified.mtimeMs }),
-            )
-        }
-        const prismaStatus = await fs.readJson(join(tempDir, `blitz-${creationDate.birthtimeMs}-prisma-client.json`))
-        const lastModified = await fs.stat(join(process.cwd(), "db/schema.prisma"))
-        if (prismaStatus.lastModified !== lastModified.mtimeMs) {
-            let prismaSpinner = log.spinner(`Generating Prisma client`).start()
-            const result = await runPrisma(["generate"], true)
-            if (result.success) {
-                prismaSpinner.succeed(log.greenText("Generated Prisma client"))
-            } else {
-                prismaSpinner.fail()
-                console.log("\n" + result.stderr)
-                process.exit(1)
-            }
-            await fs.writeJson(join(process.cwd(), `blitz-${creationDate.birthtimeMs}-prisma-client.json`), {
-                lastModified: lastModified.mtimeMs,
-            })
+      const foundPrismaClient = resolveFrom.silent(process.cwd(), "@prisma/client")
+      if (
+        !foundPrismaClient ||
+        (foundPrismaClient && !fs.existsSync(join(foundPrismaClient, "../../..", ".prisma")))
+      ) {
+        let prismaSpinner = log.spinner(`Generating Prisma client`).start()
+        const result = await runPrisma(["generate"], true)
+        if (result.success) {
+          prismaSpinner.succeed(log.greenText("Generated Prisma client"))
         } else {
-            log.success("Prisma client was up to date")
+          prismaSpinner.fail()
+          console.log("\n" + result.stderr)
+          process.exit(1)
         }
+      } else {
+        log.success("Prisma client was up to date")
```

Comment:
I'm not sure about this message. According to the code, prisma client **was found**, and not necessarily **up to date**. Saying "up to date" can be misleading.

Reply:
If prisma client was found, then we're skipping the `prisma generate` step. IMO we shouldn't print anything in this case.

Reply:
Ya very valid point